src/components/common/PasswordInput/PasswordInput.test.tsx

```diff
@@ -0,0 +1,62 @@
+import { screen, fireEvent } from '@testing-library/react';
+import { render } from "../../../test/utils";
+import '@testing-library/jest-dom';
+import { describe, it, expect } from 'vitest';
+import { PasswordInput } from '.';
+
+describe('PasswordInput Component', () => {
+    const mockOnPasswordChange = vi.fn();
+
+    const setup = () => {
+        render(
+            <PasswordInput
+                password="testPassword"
+                onPasswordChange={mockOnPasswordChange}
+                name="password"
+                id="password"
+                placeholder="Enter your password"
+            />
+        );
+    };
+
+    it('should render the component', () => {
+        setup();
+        const input = screen.getByPlaceholderText('Enter your password');
+        expect(input).toBeInTheDocument();
+    });
+
+    it('should toggle password visibility', () => {
+        setup();
+        const toggleButton = screen.getByRole('button', { name: /show password/i });
+        const input = screen.getByPlaceholderText('Enter your password');
+
+        // Password should be hidden initially
+        expect(input).toHaveAttribute('type', 'password');
+
+        // Click to show the password
+        fireEvent.click(toggleButton);
```

Comment:
Consider using the [userevent api](https://testing-library.com/docs/user-event/intro/) instead